### PR TITLE
Allow additional bahro stone type GUI

### PR DESCRIPTION
**Pull request by Hazado**
This changes allow additional bahro-style linking panels by making the code that calls them depend on variables from ``xLinkingBookDefs.py``

Ive already changed ``xLinkingBookDefs.py`` for the Bahro panels to include what GUI they use as well and the name used for Share/NoShare graphic.

When a new single book panel style is added, the GUI name would need to be added to ``xBookGUIs.py`` to load it ingame, and the share/noshare graphic added to ``xLinkingBookDefs.py``.

### CODE REVIEW

File: Scripts/Python/xLinkingBookGUIPopup.py [hunk 1]
```diff
@@ -798,15 +798,15 @@ def IShowBahroBook(self,linkingPanel):
         agePanel = TargetAge.value
         # share book image is on the "first" page while the linking panel is on the "second" page
         if PtIsSinglePlayerMode() or OfferedBookMode:
-            bookdef = '<font size=10>' + xLinkingBookDefs.BahroNoShare +'<pb>' + xLinkingBookDefs.TransLinkStart + linkingPanel + xLinkingBookDefs.LinkEnd
+            bookdef = '<font size=10>' + getattr(xLinkingBookDefs,shareName+'NoShare') +'<pb>' + xLinkingBookDefs.TransLinkStart + linkingPanel + xLinkingBookDefs.LinkEnd
```
Comment (Hoikas) [inline Scripts/Python/xLinkingBookGUIPopup.py:801]:
```suggestion
            bookdef = '<font size=10>' + getattr(xLinkingBookDefs, f'{shareName}NoShare') +'<pb>' + xLinkingBookDefs.TransLinkStart + linkingPanel + xLinkingBookDefs.LinkEnd
```

File: Scripts/Python/xLinkingBookGUIPopup.py [hunk 2]
```diff
@@ -798,15 +798,15 @@ def IShowBahroBook(self,linkingPanel):
...
         else:
-            bookdef = '<font size=10>' + xLinkingBookDefs.BahroShare +'<pb>' + xLinkingBookDefs.TransLinkStart + linkingPanel + xLinkingBookDefs.LinkEnd
+            bookdef = '<font size=10>' + getattr(xLinkingBookDefs,shareName+'Share') +'<pb>' + xLinkingBookDefs.TransLinkStart + linkingPanel + xLinkingBookDefs.LinkEnd
```
Comment (Hoikas) [inline Scripts/Python/xLinkingBookGUIPopup.py:803]:
```suggestion
            bookdef = '<font size=10>' + getattr(xLinkingBookDefs, f'{shareName}Share')  +'<pb>' + xLinkingBookDefs.TransLinkStart + linkingPanel + xLinkingBookDefs.LinkEnd
```

**Review by Hoikas — COMMENTED** [review]
I think this otherwise looks good. I want to do a careful search through the scripts to make sure that there aren't any other spots where someone is making assumptions about `xAgeLinkingBooks`.

**Comment by Hazado** [comment]
> I think this otherwise looks good. I want to do a careful search through the scripts to make sure that there aren't any other spots where someone is making assumptions about `xAgeLinkingBooks`.

Only used in two Ahnonay age script for the Ahnonay and AhnonayCathedral panels.
I haven't found anywhere in the python where `xAgeLinkingBooks` is used otherwise.

**Review by Hoikas — APPROVED** [review]

**Pull request merged**